wasm/src/api.rs

```diff
@@ -40,3 +53,82 @@ pub fn run_clvm(program: &[u8], args: &[u8]) -> Vec<u8> {
         Err(_eval_err) => format!("{:?}", _eval_err).into(),
     }
 }
+
+#[wasm_bindgen]
+/**
+ * Return serialized result of clvm program with cost
+ *
+ * cost will be available at the first 8 bytes of returned Vec<u8>(=Uint8Array).
+ * bytes of node will be available from 8th byte offset of returned Vec<u8>.
+ *
+ * @example
+ * const result = run_chia_program(...); // Uint8Array
+ * const cost = vec_u8_to_u64(result.subarray(0, 8)) // BigInt;
+ * const serialized_node = result.subarray(8); // Uint8Array
+ */
+pub fn run_chia_program(
+    program: &[u8],
+    args: &[u8],
+    max_cost: Cost, // Expecting `BigInt` to be passed from JavaScript world
+    flag: u32,
+) -> Vec<u8> {
+    let mut allocator = Allocator::new();
+    let program = node_from_bytes(&mut allocator, program).unwrap();
+    let args = node_from_bytes(&mut allocator, args).unwrap();
+    let dialect = ChiaDialect::new(flag);
+
+    let r = run_program(
+        &mut allocator,
+        &dialect,
+        program,
+        args,
+        max_cost,
+        None,
+    );
+    match r {
+        Ok(reduction) => {
+            let cost = reduction.0;
+            let node_bytes = node_to_bytes(
+                &Node::new(&allocator, reduction.1)
+            ).unwrap();
+
+            [
+                u64_to_vec_u8(cost),
+                node_bytes,
+            ].concat()
+        },
+        Err(_eval_err) => format!("{:?}", _eval_err).into(),
+    }
+}
+
+#[wasm_bindgen]
+pub fn u64_to_vec_u8(num: u64) -> Vec<u8> {
```

Comment:
instead of this function, you could just use:

```
cost.to_be_bytes().to_vec()
```

Reply:
Thanks! It's working!